backends-common/rabbitmq/src/main/java/org/apache/james/backend/mailqueue/RabbitMQConfiguration.java

```diff
@@ -0,0 +1,73 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.backend.mailqueue;
+
+import java.util.Objects;
+
+import org.apache.commons.configuration.PropertiesConfiguration;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+public class RabbitMQConfiguration {
+
+    private static final String URI = "uri";
+    private static final String MANAGEMENT_URI = "management.uri";
+
+    public static RabbitMQConfiguration from(PropertiesConfiguration configuration) {
+        String uri = configuration.getString(URI);
+        Preconditions.checkState(!Strings.isNullOrEmpty(uri), "You need to specify the URI of RabbitMQ");
+
+        String managementUri = configuration.getString(MANAGEMENT_URI);
+        Preconditions.checkState(!Strings.isNullOrEmpty(managementUri), "You need to specify the management URI of RabbitMQ");
+        return new RabbitMQConfiguration(uri, managementUri);
```

Comment:
what about checking they are actual URI by using URI object ?

Reply:
Ok, but I had to push -f in order to solve fixup conflicts.